### DISCUSSION TITLE
Rules and berry script for ESP32 ANALOG sensors

### DISCUSSION THREAD

**Discussion by gkarag** [arendst/Tasmota#13753] [Q&A] [answered]
Hello community,

I have two ADC Inputs assigned to `GPIO34` and `GPIO35`. Then I try to write a rule which acts on values of `ANALOG#A1` and `ANALOG#A2` as they appear in the SENSOR JSON. The rule never seemed to fire no matter what values I am checking on, for example:
```
Rule1 ON ANALOG#A1>0 DO Power1 0 ENDON
```

Similarly I tried to access the values from a berry script, as suggested by an example in the [Tasmota's Berry](https://tasmota.github.io/docs/Berry/#rules) page (I needed to correct syntax errors first):
```
import string

def rule_adc(value, trigger)
  var i=string.find(trigger,"#A")
  var tr=string.split(trigger,i+2)
  var adc=number(tr[1])
  print("value of adc",adc," is ",value)
end

tasmota.add_rule("ANALOG#A1",rule_adc)
tasmota.add_rule("ANALOG#A2",rule_adc)
```
The example kind of works. Kind of because it logs values every 5 minutes and I would like it do so more frequently.

Any suggestions what should I do?

**Comment by barbudor** [comment, marked as answer]
Maybe activate the rule with `Rule1 1` ?

**Reply by gkarag** [reply]
Thanks for the reply. I made sure the rule is enabled with `Rule1 1` but unfortunately it didn't help.

**Reply by barbudor** [reply]
Please provide the output of `Rule0` and a sample of your `tele/.../SENSOR` message

**Reply by gkarag** [reply]
Here it is:
```
22:27:47.278 CMD: rule0
22:27:47.285 RSL: RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":29,"Free":482,"Rules":"ON ANALOG#A1>0 Power1 0 ENDON"}}
22:27:47.293 RSL: RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
22:27:47.300 RSL: RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
```

```
22:35:59.300 CMD: status 8
22:35:59.441 RSL: STATUS8 = {"StatusSNS":{"Time":"2021-11-21T22:35:59","ANALOG":{"A1":4095,"A2":4095},"ESP32":{"Temperature":53.3},"TempUnit":"C"}}
```

Currently I use the officially released Tasmota 10.0.0

**Reply by barbudor** [reply]
Wrong syntax on Rule1
You are missing the `DO`

**Reply by gkarag** [reply]
That was it 🤦‍♂️ Thanks a lot!

**Comment by sfromis** [comment]
https://github.com/arendst/Tasmota/pull/13756 should provide the ability to read sensors from Berry, instead of relying on the periodic triggers.

**Reply by Littlesloths** [reply]
shouldn't the rule look like:
StatusSNS#ANALOG#A1
according to rule syntax?

**Reply by sfromis** [reply]
That's only when using values from the JSON output of `Status 10`, where the sensor data is wrapped in a StatusSNS object, not when using the JSON structure part of the regular /SENSOR payloads - that format is also used continuously outside of the TelePeriod publishing.
